x-pack/plugins/observability_ai_assistant/public/components/insight/actions_menu.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState } from 'react';
+import { EuiButtonIcon, EuiContextMenu, EuiPanel, EuiPopover } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { UseGenAIConnectorsResult } from '../../hooks/use_genai_connectors';
+import { ConnectorSelectorBase } from '../connector_selector/connector_selector_base';
+
+export function ActionsMenu({
+  connectors,
+  onEditPrompt,
+}: {
+  connectors: UseGenAIConnectorsResult;
+  onEditPrompt: () => void;
+}) {
+  const [isPopoverOpen, setPopover] = useState(false);
+
+  const onButtonClick = () => {
+    setPopover(!isPopoverOpen);
```

Comment:
nit: when performing operations on the current state value it's safer to use the callback approach. I don't think it makes an actual difference in this case (at the moment at least) because `setPopover` is only called once so consider this an optional nit.

```suggestion
    setPopover(value => !value);
```

https://react.dev/learn/queueing-a-series-of-state-updates

Reply:
Interesting read thanks!